src/Resources/contao/dca/tl_survey.php

```diff
@@ -93,5 +93,16 @@
                 // Keep this for MySQL Strict mode. Otherwise Contao would save an empty string
                 'sql' => ['type' => 'boolean', 'default' => false],
             ],
+            'store_records' => [
+                'inputType' => 'checkbox',
+                'filter' => true,
+                'eval' => [
+                    'tl_class' => 'w50 m12',
+                ],
+                'default' => true,
+                'save_callback' => [static fn ($v) => '1' === $v],
+                // Keep this for MySQL Strict mode. Otherwise Contao would save an empty string
+                'sql' => ['type' => 'boolean', 'default' => false],
```

Comment:
default value should match `default` from DCA definition

```suggestion
                'sql' => ['type' => 'boolean', 'default' => true],
```